teps/0077-pipeline-partial-execution.md

```diff
@@ -0,0 +1,496 @@
+          - name: print-the-recipe 
+    ```
+
+4. **Using skip tasks with output resources**
+   Allow users to specify a list of PipelineTasks to be skipped when defining a
+   PipelineRun, and include resource overrides to handle dependent Tasks. Some
+   subsequent Tasks may be dependent on the disabled Tasks, that is, they expect
+   some resources from the disabled Tasks. The user can optionally make those
+   resources available by providing access to the workspaces and expected
+   results in the `disabledTasks` definition as such:
+   
+    ```yaml
+    apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+        name: example-pipeline-run
+    spec:
+        pipelineRef:
+            name: example-pipeline
+        workspaces:
+            - name: shared-data
+        persistentVolumeClaim:
+            claimName: shared-task-storage
+        disabledTasks:
+            - name: fetch-the-recipe
+              outputs:
+              workspaces:
+                - name: filedrop
+                  workspace: shared-data
+                  results:
+                - name: status
+                  value: $(foo.status.results)
+            - name: print-the-recipe
+              workspaces:
+              results:
+    ```
+   
+5. **Resume from Tasks in a previous PipelineRun**
+   We can implement some way to allow a PipelineRun to pick up from a previous
+   PipelineRun, and resume from specific Tasks. Similar to Jenkins’ restart from
+   stage.
+   
+    ```yaml
+    apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      name: example-pipeline-run
+    spec:
+      pipelineRef:
+        name: example-pipeline
+        fromPreviousRun: foo
+        resumeFrom: [print-the-recipe]
+    ```
+   
+    One of the challenge here is, a user has to figure out a DAG. As point
+    of resume can be plural.
+   
+## Infrastructure Needed (optional)
+
+<!--
+Use this section if you need things from the project/SIG.  Examples include a
+new subproject, repos requested, github details.  Listing these here allows a
+SIG to get the process for these resources started right away.
+-->
+
+## Upgrade & Migration Strategy (optional)
+
+<!--
+Use this section to detail wether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
+
+## Implementation Pull request(s)
+
+<!--
+Once the TEP is ready to be marked as implemented, list down all the Github
+Pull-request(s) merged.
+Note: This section is exclusively for merged pull requests, for this TEP.
+It will be a quick reference for those looking for implementation of this TEP.
+-->
+
+## References (optional)
+
+1. [Google Doc: Disabling a Task in a Pipeline](https://docs.google.com/document/d/1rleshixafJy4n1CwFlfbAJuZjBL1PQSm3b0Q9s1B_T8/edit#heading=h.jz9jia3av6h1)
+2. [TEP-0065](https://github.com/tektoncd/community/pull/422)
```

Comment:
drive by request: it'd be great to include a link to https://github.com/tektoncd/pipeline/issues/50 as well - we're addressing our oldest open issue in pipelines!! :D